matplotlib2tikz/save.py

```diff
@@ -309,6 +309,8 @@ def _recurse(data, obj):
             content.extend(cont, child.get_zorder())
         elif isinstance(child, mpl.legend.Legend):
             data = legend.draw_legend(data, child)
+            if data['legend colors']:
+                content.extend(data['legend colors'], 0)
```

Comment:
It works. addlegendimage has to be at the beginning, thats why zorder has to be 0